docs/03-Development-Process.md

```diff
@@ -12,13 +12,64 @@ examples :
 - feature/issue-15469
 - bugfix/issue-15469
 
+Origin vs Upstream in a Forked GitHub Workflow:
+
+| Term         | What it means                                               | Typical URL example                                | Used for                                         |
+| ------------ | ----------------------------------------------------------- | -------------------------------------------------- | ------------------------------------------------ |
+| **origin**   | Your fork of the repository (remote on your GitHub account) | `https://github.com/your-username/together.git`    | Pushing your rebased changes, creating PRs       |
+| **upstream** | The original repository you forked from                     | `https://github.com/Together-100Devs/together.git` | Pulling the latest changes from the main project |
+
 Developers should follow the following steps :
 
-- Updated the `main` : `git pull origin main`
-  - It is generally advised to keep your branch consistently up to date with main
-- create feature branch : `git checkout -b feature/issue-15469`
+- Create a feature branch : `git checkout -b feature/issue-15469`
 - Develop the specified requirements
-- commit and Push the code
+- Commit locally:
+
+  - stage your chages: `git add .`
+  - commit your changes: `git commit -m "your commit message"`
+
+- It is generally advised to keep your branch consistently up to date with `main`.
+
+  - Grab the latest upstream version of main, in case main had changes: `git fetch upstream`
+
+  - upstream = grabbing all upstream branches (upstream/main, upstream/add-dependabot, ect)
+
+- Rebase your feature branch on top of the latest upstream/main. We're telling git to use the "main" branch from upstream as the new "base" for your feature branch `git rebase upstream/main`
+
+- To keep the commit history clean, we require that you squash your commits into a single commit using interactive rebase:
+
+  - This shows all commits since you branched off of main: `git rebase -i upstream/main`
+
+  - Your editor will open up with a list of commits (oldest first). New to using the vim editor? Look below for a cheatsheet
+
+  > pick c32423434 first commit
+  > pick af243df second commit
+  > pick 234asf45 third commit
+
+  - Keep the first commit as pick, change the rest to squash (or s):
+
+    > pick c32423434 first commit
+    > squash af243df second commit
+    > squash 234asf45 third commit
+
+  - Save and close the editor, you'll be prompted to edit the final commit message (which starts off as all of the previous messages combined)
+
+  - optional: `git log --oneline`: if you want to double check that it merged into one commit. Or on upper left side of VSCode click on source control to see the history (the icon with 3 circles, connected by 2 lines)
```

Comment:
could we add a screen shot here?

Reply:
definitely! I somehow completely missed the asset folder you all had earlier haha